xgboost.ipynb

```diff
@@ -0,0 +1,227 @@
+    "X = da.random.normal(size=(n, d), chunks=n // 100)\n",
+    "w_star = da.random.uniform(size=d, chunks=d)\n",
+    "w_star = w_star**3\n",
+    "y = da.sign(X @ w_star)\n",
+    "\n",
+    "# for binary:logistic objective, only [0, 1] labels allowed\n",
+    "y = (y + 1) / 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's separate into a training set and testing set, which will allow for good evaluation after training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.15)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's try and predict from this data with [dask-xgboost][dxgb]:\n",
+    "\n",
+    "[dxgb]:https://github.com/dask/dask-xgboost"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask_xgboost as dxgb\n",
+    "import xgboost as xgb"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "dask-xgboost is a small wrapper around xgboost, and will behave the same as xgboost.\n",
+    "\n",
+    "During training it will take care of handling the data, and allow xgboost to use it's own distributed scheduler."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "params = {'objective': 'binary:logistic', 'nround': 1000, \n",
+    "          'max_depth': 5, 'eta': 0.01, 'subsample': 0.5, \n",
+    "          'min_child_weight': 0.5}\n",
+    "\n",
+    "bst = dxgb.train(client, params, X_train, y_train)\n",
+    "bst"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `bst` object is a regular `xboost.Booster` object, and has all the familar methods available. For example,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import pandas as pd\n",
+    "\n",
+    "if isinstance(w_star, da.Array):\n",
+    "    w_star = w_star.compute()\n",
+    "idx_top = np.argsort(w_star)[-7:]\n",
+    "top = w_star[idx_top]\n",
+    "\n",
+    "fig, axs = plt.subplots(figsize=(12, 8), ncols=2)\n",
+    "axs[0] = xgb.plot_importance(bst, ax=axs[0], height=0.8, max_num_features=7)\n",
+    "axs[0].grid(False, axis=\"y\")\n",
+    "axs[0].set_title('Estimated feature importance')\n",
+    "\n",
+    "df = pd.DataFrame({'feature_importance': top}, index=idx_top)\n",
+    "df.plot.bar(ax=axs[1])\n",
+    "axs[1].set_title('Ground truth feature importance')\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Or we can plot the ROC curve after we do some predictions:"
```

Comment:
As someone who is a novice in machine learning I have to admit that I don't have a strong understanding of ROC curves.  It might be good to spell out the acronym once in the style of `Three Letter Acronym (TLA)` and then state that "we see that results are good because the curve tends towards the upper left", just to give the novice reader the ability to appreciate how well things worked out.

Reply:
I've added an explanation for what ROC curves are.

Reply:
And a link too.